core/src/main/scala/com/codacy/analysis/core/storage/FileDataStorage.scala

```diff
@@ -0,0 +1,76 @@
+package com.codacy.analysis.core.storage
+
+import better.files.File
+import io.circe._
+import io.circe.syntax._
+import org.log4s.{Logger, getLogger}
+
+abstract class FileDataStorage[T] {
+  private val logger: Logger = getLogger
+
+  implicit val encoder: Encoder[T]
+  implicit val decoder: Decoder[T]
+
+  def storageFilename: String
+
+  val storageFile: File = File.currentWorkingDirectory / ".codacy" / "codacy-analysis-cli" / storageFilename
+
+  def compare(current: T, value: T): Boolean
+
+  private def writeToFile(file: File, content: String): Boolean =
+    synchronized {
+      file.createFileIfNotExists(createParents = true)
+      file.write(content)
+      file.contentAsString == content
```

Comment:
Why?

Reply:
Now using `Try` and not making this validation.